ice/src/main/java/com/altinity/ice/cli/internal/metrics/InsertWatchMetrics.java

```diff
@@ -0,0 +1,263 @@
+ *
+ * <p>This class uses a singleton pattern because Prometheus metrics can only be registered once per
+ * JVM.
+ */
+public class InsertWatchMetrics {
+
+  private static final Logger logger = LoggerFactory.getLogger(InsertWatchMetrics.class);
+
+  // Singleton instance
+  private static volatile InsertWatchMetrics instance;
+  private static final Object lock = new Object();
+
+  // ==========================================================================
+  // Metric Names
+  // ==========================================================================
+
+  private static final String LABEL_TABLE = "table";
+  private static final String LABEL_QUEUE = "queue";
+
+  private static final String[] WATCH_LABELS = {LABEL_TABLE, LABEL_QUEUE};
+
+  // Messages/Files processed
+  private static final String MESSAGES_RECEIVED_TOTAL_NAME = "ice_watch_messages_received_total";
+  private static final String MESSAGES_RECEIVED_TOTAL_HELP =
+      "Total number of SQS messages received";
+
+  private static final String EVENTS_RECEIVED_TOTAL_NAME = "ice_watch_events_received_total";
+  private static final String EVENTS_RECEIVED_TOTAL_HELP =
+      "Total number of S3 events received (one message may contain multiple events)";
+
+  private static final String EVENTS_MATCHED_TOTAL_NAME = "ice_watch_events_matched_total";
+  private static final String EVENTS_MATCHED_TOTAL_HELP =
+      "Total number of S3 events that matched the pattern";
+
+  private static final String EVENTS_NOT_MATCHED_TOTAL_NAME = "ice_watch_events_not_matched_total";
+  private static final String EVENTS_NOT_MATCHED_TOTAL_HELP =
+      "Total number of S3 events that did not match any input pattern";
+
+  private static final String EVENTS_SKIPPED_TOTAL_NAME = "ice_watch_events_skipped_total";
+  private static final String EVENTS_SKIPPED_TOTAL_HELP =
+      "Total number of S3 events skipped (non-ObjectCreated events)";
+
+  // Files inserted
+  private static final String FILES_INSERTED_TOTAL_NAME = "ice_watch_files_inserted_total";
+  private static final String FILES_INSERTED_TOTAL_HELP =
+      "Total number of files successfully inserted into the catalog";
+
+  // Transactions
+  private static final String TRANSACTIONS_TOTAL_NAME = "ice_watch_transactions_total";
+  private static final String TRANSACTIONS_TOTAL_HELP =
+      "Total number of insert transactions committed";
+
+  private static final String TRANSACTIONS_FAILED_TOTAL_NAME =
+      "ice_watch_transactions_failed_total";
+  private static final String TRANSACTIONS_FAILED_TOTAL_HELP =
+      "Total number of insert transactions that failed";
+
+  // Retry state
+  private static final String RETRY_ATTEMPTS_TOTAL_NAME = "ice_watch_retry_attempts_total";
+  private static final String RETRY_ATTEMPTS_TOTAL_HELP =
+      "Total number of retry attempts due to failures";
+
+  // SQS errors
+  private static final String SQS_RECEIVE_ERRORS_TOTAL_NAME = "ice_watch_sqs_receive_errors_total";
```

Comment:
should we perhaps drop sqs from metric names here and below for consistency with metrics above + in case we decide to add support for other message queue?